src/main/java/de/danielbechler/diff/identity/IdentityStrategyResolver.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2014 Daniel Bechler
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.danielbechler.diff.identity;
+
+import de.danielbechler.diff.node.DiffNode;
+
+public interface IdentityStrategyResolver {
+	IdentityStrategy resolveIdentityStrategy(DiffNode node);
+	IdentityStrategy resolveByCollectionElement(Object collectionElement);
```

Comment:
No, mine is `ofTypeAndProperty(Class<?> type, String... propertyNames)`.

The `ofNode()` is the one I haven't been able to use at all because of the map's key on the path:
```java
           .ofNode(
                    // this is not very useful without wildcards on maps and collections...
                    NodePath.startBuilding().propertyName("productMap").mapKey("PROD1")
                            .propertyName("productVersions").build()
                )
```